grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import org.curioswitch.common.protobuf.json.MessageMarshaller;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+public final class UnframedGrpcErrorHandlerBuilder {
+
+    private static final Set<UnframedGrpcErrorResponseType> DEFAULT_RESPONSE_TYPES =
+            ImmutableSet.copyOf(UnframedGrpcErrorResponseType.values());
+
+    @Nullable
+    private UnframedGrpcStatusMappingFunction statusMappingFunction;
+
+    @Nullable
+    private MessageMarshaller jsonMarshaller;
+
+    private Set<UnframedGrpcErrorResponseType> responseTypes = DEFAULT_RESPONSE_TYPES;
+
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        this.jsonMarshaller = requireNonNull(jsonMarshaller, "jsonMarshaller");
+        return this;
+    }
```

Comment:
Should we provide this message marshaller as default value here?
https://github.com/line/armeria/blob/42edbb2940a0447f8df9dd189fe24acfc842335a/grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlers.java#L72-L85

Reply:
And should we provide a method to register custom `Message` like this?

```java
public UnframedGrpcErrorHandlerBuilder registerMessageToMarshaller(Message message) {
    requireNonNull(message, "message");
    jsonMarshaller = jsonMarshaller.toBuilder().register(message).build();
    return this;
}
```